lib/git_reflow.rb

```diff
@@ -107,53 +108,58 @@ def deliver(options = {})
-            always_deploy_and_cleanup = GitReflow::Config.get('reflow.always-deploy-and-cleanup') == "true"
-            deploy_and_cleanup = always_deploy_and_cleanup || (ask "Would you like to push this branch to your remote repo and cleanup your feature branch? ") =~ /^y/i
-
-            if deploy_and_cleanup
-              run_command_with_label "git push origin #{base_branch}"
-              run_command_with_label "git push origin :#{feature_branch}"
-              run_command_with_label "git branch -D #{feature_branch}"
-              puts "Nice job buddy."
-            else
-              puts "Cleanup halted.  Local changes were not pushed to remote repo.".colorize(:red)
-              puts "To reset and go back to your branch run \`git reset --hard origin/#{base_branch} && git checkout #{feature_branch}\`"
-            end
+
+          # displays the current status and prompts user for confirmation
+          self.status base_branch
+          always_deliver = GitReflow::Config.get('reflow.always-deliver') == "true"
+          deliver = always_deliver || (ask "This is the current status of your Pull Request. Are you sure you want to deliver? ") =~ /^y/i
+          
+          if deliver
+            puts "Merging pull request ##{existing_pull_request.number}: '#{existing_pull_request.title}', from '#{existing_pull_request.feature_branch_name}' into '#{existing_pull_request.base_branch_name}'"
+
+            begin
+              committed = existing_pull_request.merge!(destination_branch: base_branch, force: options['skip_lgtm'], message: commit_message)
+
+              if committed
+                say "Pull Request successfully merged.", :success
+
+                # check if user always wants to push and cleanup, otherwise ask
+                always_cleanup = GitReflow::Config.get('reflow.always-cleanup') == "true"
+                cleanup = always_cleanup || (ask "Would you like to cleanup your feature branch? ") =~ /^y/i
+
+                if cleanup
+                  # Pulls merged changes from remote base_branch
+                  run_command_with_label "git pull origin #{base_branch}"
+                  run_command_with_label "git push origin #{base_branch}"
+                  run_command_with_label "git push origin :#{feature_branch}"
+                  run_command_with_label "git branch -D #{feature_branch}"
+                  puts "Nice job buddy."
+                else
+                  puts "Cleanup halted.  Local changes were not pushed to remote repo.".colorize(:red)
+                  puts "To reset and go back to your branch run \`git reset --hard origin/#{base_branch} && git checkout #{feature_branch}\`"
+                end
+              else
+                say "There were problems commiting your feature... please check the errors above and try again.", :error
+              end
+            rescue GitReflow::GitServer::MergeError => error
+                say error.inspect, :deliver_halted
```

Comment:
Let's make this `error.to_s`.  [StandardError#to_s](http://ruby-doc.org/core-2.3.1/Exception.html#method-i-to_s) will report the exception's `message` by default.